src/rcc/mod.rs

```diff
@@ -228,37 +228,30 @@ impl Rcc {
 
         let pll_freq = freq / (pll_cfg.m as u32) * (pll_cfg.n as u32);
         let r = (pll_freq / (pll_cfg.r as u32)).Hz();
-        let q = match pll_cfg.q {
-            Some(div) if div > 1 && div <= 8 => {
-                self.pllsyscfgr
-                    .write(move |w| unsafe { w.pllq().bits(div - 1) });
+        let mut q = None;
+        let mut p = None;
+
+        self.pllsyscfgr.write(|w| unsafe {
+            w.pllsrc().bits(pll_sw_bits);
+            w.pllm().bits(pll_cfg.m - 1);
+            w.plln().bits(pll_cfg.n);
+            w.pllr().bits(pll_cfg.r - 1);
+            w.pllren().set_bit();
+            if let Some(div) = pll_cfg.q {
+                assert!(div > 1 && div <= 8);
```

Comment:
@dotcypress we haven't used asserts so far, what is your take on these?
I vaguely remember that they might increase flash size quite significantly.

Reply:
There's a couple of asserts already at the start of the method. I'd expect that if you're building with LTO and using constants values in your PllConfig, the compiler can probably optimize the asserts away. That said, I'd be happy to change these back to if-statements like they were in the old code. It just seemed odd to me that if you supplied an invalid config, it just ignored the bits that were invalid and kept going.

Reply:
@davidlattimore totally agree, this approach is much better